### PR TITLE
Add support for Node 10

### LINKED ISSUE #7184: Support Node 10
Fabric currently doesn't build properly with Node 10. Since Node 10 is now LTS, we should work on supporting it. 

More info copied from an email from @cliffkoh:

> I just gave Node 10@LTS a whirl and NPM 5.4 does not work well with Node 10 (some OUFR dependencies literally complained that we're using NPM 5.4 with Node 10). Upgrading to NPM 5.6 resulted in a nasty warning about NPM 5.6 being incompatible with Node 10 (even without doing a proper rush install) and NPM 6.4 is entirely busted. I'm guessing the path to staying current would require us to move to yarn or pnpm since there [NPM 6.4 has blocking issues with Rush](https://github.com/Microsoft/web-build-tools/issues/886)...

See #7185 for investigation of switching to pnpm or yarn.

### PR DESCRIPTION
**Pull request by ecraig12345**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #7184
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

Node 10 is now LTS, so we should support it in Fabric. We will also continue to support Node 8 for now.

#### Scripts to test
Checked items work; unchecked items are broken or untested.
- [x] build (also includes basic tests)
- [x] start
- [x] start-exp
- [x] code-style
- [x] bundlesize
- [x] create-component
- [x] create-package
- [x] update-api
- [x] update-snapshots
- [x] check-for-changed-files
- [x] prettier
- [x] generate-version-files
- [x] codepen
- [x] precommit
- [x] (apps/vr-tests) screener
- [x] (apps/vr-tests) screener:local
- [x] (apps/vr-tests) start

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/7264)

### CODE REVIEW

File: rush.json [hunk 1]
```diff
@@ -92,7 +92,7 @@
    * Specify a SemVer range to ensure developers use a NodeJS version that is appropriate
    * for your repo.
    */
-  "nodeSupportedVersionRange": ">=8.0.0 <10.0.0",
+  "nodeSupportedVersionRange": ">=8.0.0 <9.0.0 || >=10.0.0 <11.0.0",
```
Comment (cliffkoh) [inline rush.json]: For node 10, maybe restrict it to more recent versions...

Comment (dzearing) [reply rush.json]: what do you mean?

Comment (dzearing) [reply rush.json]: or... do you know of problems with some versions?

Comment (ecraig12345) [reply rush.json]: What do you think a good start version would be? I'm not sure how to determine that.

Comment (dzearing) [reply rush.json]: gotcha; maybe just start at the current LTS then? (10.14.1) as we're just starting to test now.

There were a few security updates recommending minbar 10.14.0, and the .1 patch was specifcally for windows msi issues.

https://nodejs.org/en/blog/vulnerability/november-2018-security-releases/

Comment (pgonzal) [reply rush.json]: > What do you think a good start version would be? I'm not sure how to determine that.

(1) an LTS version, and (2) something you've actually tested yourself.

It is super frustrating to waste an hour investigating somebody's issue, only to find out that it was just a buggy build of NodeJS.  We got burned by this repeatedly until we finally started insisting on LTS versions.

Comment (KevinTCoughlin) [reply rush.json]: Recommended DL https://groups.google.com/forum/#!forum/nodejs-sec

**Pull request merged**